pkg/k8s/node.go

```diff
@@ -63,5 +63,36 @@ func ParseNode(k8sNode *v1.Node) *node.Node {
 		}
 	}
 
+	// Spec.PodCIDR takes precedence since it's
+	// the CIDR assigned by k8s controller manager
+	// In case it's invalid or empty then we fall back to our annotations.
+	if node.IPv4AllocCIDR == nil {
```

Comment:
Ah, I didn't think the user would set the annotations manually, ok on that case it makes sense to switch them precedence order.

What happens if he changes the CIDR after containers have started in the node? The old containers will keep their IP addresses and the new containers will get a new IP addresses, will cilium capable of routing for both CIDRs?